### PR TITLE
Acs setup

### PR DESCRIPTION
**Pull request by shawn-phy**
This Pr aims to add the functionality of the ACS712 current sensor to the modules code. 
It also creates a thread for reading voltage in the main file.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -23,7 +23,8 @@ static int protothreadReadDHT(struct pt *pt)
     PT_WAIT_UNTIL(pt, millis() - lastTimeread > interval);
     float temp = readtemp();
     float humid = readhumidity(); 
-    Serial.printf("temp: %.3f, humid: %.3f\n ", temp, humid);
+    Serial.printf("temp: %.3f, humid: %.3f\n", temp, humid);
```
Comment (BarbraGitonga) [inline src/main.cpp:26]: why not use ets_printf() function here too?

File: src/main.h [hunk 2]
```diff
@@ -10,6 +10,7 @@
 #include "arduino_secrets.h"
 #include "WiFi.h"
 #include "PIR/pir.h"
```
Comment (BarbraGitonga) [inline src/main.h]: Add the new libraries to the include directory instead of giving the path in #include

**Review by BarbraGitonga — APPROVED** [review]
Fix the changes before merging, Love the arrangement of the script.

File: src/main.cpp (hunk 1 above)
Comment (shawn-phy) [reply src/main.cpp:26]: **Read** the documentation  from the esp IDF dear, ``` C
/**
  * @brief  Printf the strings to uart or other devices, similar with printf, simple than printf.
  *  Can not print float point data format, or longlong data format.
  *  So we maybe only use this in ROM.
  *
  * @param  const char *fmt : See printf.
  *
  * @param  ... : See printf.
  *
  * @return int : the length printed to the output device.
  */
int ets_printf(const char *fmt, ...);```

File: src/main.h (hunk 2 above)
Comment (shawn-phy) [reply src/main.h]: @BarbraGitonga This has been handled, but I generally prefer including them with their paths, so I don't have to make edits to the same file twice.

File: src/main.cpp (hunk 1 above)
Comment (BarbraGitonga) [reply src/main.cpp:26]: Noted. Thanks

**Pull request merged**